grobid-core/src/main/java/org/grobid/core/utilities/GrobidProperties.java

```diff
@@ -312,6 +311,23 @@ private void init() {
         loadCrfEngine();
     }
 
+    /** Return the distinct values of all the engines that are needed */
+    public static Set<GrobidCRFEngine> getDistinctModels() {
+        List<GrobidCRFEngine> modelSpecificEngines = getModelSpecificEngines();
+        modelSpecificEngines.add(getGrobidCRFEngine());
```

Comment:
minor: shame that core Java doesn't seem to have a functional union method. Could use Guava's [Sets.union](https://guava.dev/releases/18.0/api/docs/com/google/common/collect/Sets.html#union(java.util.Set,%20java.util.Set)).
Otherwise I'd perhaps prefer to create the result HashSet first and then modify it (i.e. not modify the result list from another method).